packages/extension/src/ui/features/actions/ApproveTransactionScreen.tsx

```diff
@@ -50,7 +50,20 @@ export const ApproveTransactionScreen: FC<ApproveTransactionScreenProps> = ({
 }) => {
   const { t } = useTranslation()
   const navigate = useNavigate()
-  const selectedAccount = useSelectedAccount()
+  const selectedAccount0 = useSelectedAccount()
+
+  const getSignerAccount = useCallback(() => {
+    if (transactionParams.length == 1) {
+      return {
+        address: transactionParams[0].params.signerAddress,
+        networkId: transactionParams[0].params.networkId
+      }
+    }
+    return undefined
+  }, [transactionParams])
+
+  const signerAccount = useAccount(getSignerAccount())
+  const selectedAccount = signerAccount ?? selectedAccount0
```

Comment:
Is the chained tx feature affected by this issue?

Reply:
I feel that it should be fined since `selectedAccount0` is just fallback

Reply:
Yeah, now we try to select the account from the transaction param first, since when dApp builds the transaction it will use connected account to build it, so even if user switch active account in the wallet, it should still use the correct account.